include/RAJA/policy/tbb/reduce.hpp

```diff
@@ -60,12 +60,18 @@ class ReduceTBB
 
   //! constructor requires a default value for the reducer
   explicit ReduceTBB(T init_val, T initializer)
-      : data(
-            std::make_shared<tbb::combinable<T>>([=]() { return initializer; }))
   {
-    data->local() = init_val;
+    reset(init_val, initializer);
   }
 
+  //reset method should only be used by master thread
+  void reset(T init_val, T initializer)
+  {
+    data = std::shared_ptr<tbb::combinable<T>>(
+           std::make_shared<tbb::combinable<T>>([=]() { return initializer; }));
+    data->local() = init_val;
+  };
+
```

Comment:
@rhornung67 , is this what you had in mind. The downside of this is that only constructors take member initializers and I have to initialize data via a copy.